src/main/webapp/app/exam/participate/exercises/file-upload/file-upload-exam-submission.component.ts

```diff
@@ -130,7 +131,7 @@ export class FileUploadExamSubmissionComponent extends ExamSubmissionComponent i
      *  Here the new filePath, which was received from the server, is used to display the name and type of the just uploaded file.
      */
     updateViewFromSubmission(): void {
-        if (this.studentSubmission.isSynced && this.studentSubmission.filePath) {
+        if ((this.studentSubmission.isSynced && this.studentSubmission.filePath) || (this.studentSubmission.filePath && this.examTimeline)) {
```

Comment:
```suggestion
        if ((this.studentSubmission.isSynced && this.studentSubmission.filePath) || (this.examTimeline && this.studentSubmission.filePath)) {
```
Nitpick, but I think this makes the special case for the exam timeline clear more quickly to the reader